src/routes/v2/authenticated/review.ts

```diff
@@ -0,0 +1,700 @@
+
+  constructor(
+    reviewRequestService: ReviewRequestService,
+    identityUsersService: UsersService,
+    sitesService: SitesService,
+    collaboratorsService: CollaboratorsService
+  ) {
+    this.reviewRequestService = reviewRequestService
+    this.identityUsersService = identityUsersService
+    this.sitesService = sitesService
+    this.collaboratorsService = collaboratorsService
+
+    autoBind(this)
+  }
+
+  compareDiff: RequestHandler<
+    { siteName: string },
+    { items: EditedItemDto[] },
+    unknown,
+    unknown,
+    { userWithSiteSessionData: UserWithSiteSessionData }
+  > = async (req, res) => {
+    // Step 1: Check that user exists.
+    // Having session data is proof that this user exists
+    // as otherwise, they would be rejected by our middleware
+    const { userWithSiteSessionData } = res.locals
+    const { siteName } = req.params
+
+    // Check if they have access to site
+    const hasAccess = this.identityUsersService.hasAccessToSite(
+      userWithSiteSessionData.isomerUserId,
+      siteName
+    )
+
+    if (!hasAccess) {
+      return res.status(400).send()
+    }
+
+    const files = await this.reviewRequestService.compareDiff(
+      userWithSiteSessionData
+    )
+
+    return res.status(200).json({ items: files })
+  }
+
+  createReviewRequest: RequestHandler<
+    { siteName: string },
+    { pullRequestNumber: number } | ResponseErrorBody,
+    { reviewers: string[]; title: string; description: string },
+    unknown,
+    { userWithSiteSessionData: UserWithSiteSessionData }
+  > = async (req, res) => {
+    // Step 1: Check that the site exists
+    const { siteName } = req.params
+    const site = await this.sitesService.getBySiteName(siteName)
+    const { userWithSiteSessionData } = res.locals
+
+    if (!site) {
+      logger.error({
+        message: "Invalid site requested",
+        method: "createReviewRequest",
+        meta: {
+          userId: userWithSiteSessionData.isomerUserId,
+          email: userWithSiteSessionData.email,
+          siteName,
+        },
+      })
+      return res.status(404).send({
+        message:
+          "Please ensure that the site you are requesting a review for exists!",
+      })
+    }
+
+    // Step 2: Check that user exists.
+    // Having session data is proof that this user exists
+    // as otherwise, they would be rejected by our middleware
+    // Check if they are a site admin
+    const role = await this.collaboratorsService.getRole(
+      siteName,
+      userWithSiteSessionData.isomerUserId
+    )
+
+    if (!role || role !== CollaboratorRoles.Admin) {
+      logger.error({
+        message:
+          "User atttempted to create review request with invalid permissions",
+        method: "createReviewRequest",
+        meta: {
+          userId: userWithSiteSessionData.isomerUserId,
+          email: userWithSiteSessionData.email,
+          siteName,
+        },
+      })
+      return res.status(400).send({
```

Comment:
Nit: This should be 403 Forbidden.

Reply:
Hmm I think @alexanderleegs told me something about how GitHub presents stuff that you don't have access to as 400s (or was it 404s) rather than 403s, which is why I did this. 

Do we still want to enforce this behaviour?

Reply:
Yeah Github returns 404s for stuff you don't have access to - I think generally the choice for whether to return 404 or 403 works either way - 404s give potential attackers less information on whether a particular resource exists or not. I'm leaning towards 404 mostly because we already use this in a few places

Reply:
ok - for errors originating from insufficient perms, i'll return a 404? wdyt @dcshzj

Reply:
Yup 404 is fine too.